### PR TITLE
Sos 132 fix company menu

### PR DESCRIPTION
**Pull request by SeanTConrad**
This is the big PR to move the company ID into the URL, and eliminate "selected". 

Although "selected" isn't completely gone. It's still used for the company edit form, which I want to redo. I'll open a ticket.

Also includes a fix on table headers on SpendingInsights.

### CODE REVIEW

File: client/src/components/CompaniesCard.tsx [hunk 1]
```diff
@@ -22,44 +20,28 @@ export default function CompaniesCard(props:CompanyCardProps){
...
-        colors.red1000,
-        colors.red900,
-        colors.red800,
-        colors.red600,
-        colors.red400,
-        colors.red200,
-        colors.white
-    ]
+    const isSelected = (props.selected && id===props.selected.id);
+    console.log('isSelected:',isSelected);
```
Comment (aylusltd) [inline client/src/components/CompaniesCard.tsx:24]: first step to getting out of a hole?

File: client/src/components/CompaniesPage.tsx [hunk 2]
```diff
@@ -50,37 +56,89 @@ export default function CompaniesPage(){
...
+                selectCompany 
+            };
+            return CompaniesCard(props);
+        }
+        else{
+            return null;
+        }
+    });
+
+    // return (
```
Comment (aylusltd) [inline client/src/components/CompaniesPage.tsx:106]: should remove this

File: client/src/services/companies.tsx [hunk 3]
```diff
@@ -92,15 +92,8 @@ const CompaniesContext = createContext<CompaniesContextShape>(
...
-    // console.log('funsies?');
 
-    const initialize = useCallback(async()=>{
-      console.log('funsies?');
-      const companyFromStorage = localStorage.getItem('selectedCompany');
-      console.log('storage string:', companyFromStorage);
-      const companyObjFromStorage = companyFromStorage?JSON.parse(companyFromStorage):null;
-      console.log(companyObjFromStorage);
-    }, [])
+
```
Comment (aylusltd) [inline client/src/services/companies.tsx:96]: if you remove initialize here, need to remove it from line 83 as well

**Review by aylusltd — CHANGES_REQUESTED** [review]

File: client/src/components/CompaniesCard.tsx (hunk 1 above)
Comment (SeanTConrad) [reply client/src/components/CompaniesCard.tsx]: Hmm. The diff is inaccurate. I didn't add that line. The concept of "selected" is still used to launch the edit form.  I opened another ticket to fix that: https://nichols.atlassian.net/browse/SOS-137

I have a prettier MUI form overlay in my old sean-dev branch to copy over.

File: client/src/services/companies.tsx (hunk 3 above)
Comment (SeanTConrad) [reply client/src/services/companies.tsx:95]: done

File: client/src/components/CompaniesPage.tsx (hunk 2 above)
Comment (SeanTConrad) [reply client/src/components/CompaniesPage.tsx]: done

File: client/src/services/companies.tsx (hunk 3 above)
Comment (SeanTConrad) [reply client/src/services/companies.tsx:95]: Not sure why this doesn't say outdated, I committed the change.

Comment (aylusltd) [reply client/src/services/companies.tsx:95]: This line is unchanged by the commit. I added the comment in the wrong place.

**Pull request merged**